### PR TITLE
[T3A3][T13-A2]Gao Yu

### PR DESCRIPTION
**Pull request by GaoYu-Karen**

### CODE REVIEW

**Comment by GaoYu-Karen** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -55,4 +56,40 @@ public int hashCode() {
...
+}
+
+/**
+ * Takes in a Person's address in the address book by block, street, unit, postal code
+ */
+
+public interface Address{
+    public String userInput();
+}
+
```
Comment (KulaniT) [inline src/seedu/addressbook/data/person/Address.java]: You have to think here as an address has a block, street, unit and a postal code. So define the address as a combination of these. Remember to use encapsulation when defining each class.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package seedu.addressbook.data.person;
 
 import seedu.addressbook.data.exception.IllegalValueException;
+import java.util.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: don't import everything. import only what is necessary, always!

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -55,4 +56,40 @@ public int hashCode() {
...
+}
+
+public class Block implements Address{
+    public String userInput(){
+        Scanner sc = new Sanner(System.in);
+        return sc.nextString();
+    } 
+}
+
+public class Street implements Address{
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: need a space between the text and braces. take a look at the coding standards that we follow :-)

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -55,4 +56,40 @@ public int hashCode() {
...
+    public String userInput(){
+        Scanner sc = new Sanner(System.in);
+        return sc.nextString();
+    } 
+}
+
+public class Unit implements Address{
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: Please include a header comment to indicate what this class/method does?

**Comment by KulaniT** [comment]
@GaoYu-Karen ;You have to think here as an address has a block, street, unit and a postal code. So define the address as a combination of these. Remember to use encapsulation when defining each class.

**Pull request closed without merging**